### ISSUE TITLE
[BUG_SP] Unloading into american trailers

### ISSUE THREAD

**Issue by Misamangame** [Courseplay/Courseplay_FS22#2007]
Hello guys,

I seen few threads about unloader problems. For me tractors with auger wagons will not unload into vanilla american trailers. I would like to know if someone else experienced this, maybe I'm doing something wrong? Auger wagons unfold pipe and then end unloading process and disengage. If I swap this american trailer for normal one towed with tractors ( I tried 3 vanilla ones) all works without  problems. Logfile is attached, will be a bit mess as I had at that moment lots of machines running also 4 other unloaders with normal trailers for sugarbeet harvesters, apart from 3 with auger wagons for combines mentioned. Trucks are AD driven. Short video here : https://youtu.be/_2iUKNqa6vU

   
![20221013132144_1](https://user-images.githubusercontent.com/33344465/195615674-27e24d34-a9fd-46ec-9adb-14152a7294c9.jpg)

**Courseplay version**
Version used: 7.1.1.4

**Log*
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/9777230/log.txt)

**Map**
Green Valley Nebraska 4x

**Comment by pvaiko** [comment]
If you turn CP off and open the pipe at the same position, will it start discharging?

**Comment by rohne83** [comment]
Think the cause for the issue here might be a "bad" combination of grain cart's pipe discharge/raycast node and the trailers fill node. So auger wagon thinks it cannot unload as the pipe is "aiming" outside of the trailer.
Saw this happen from time to time also with completely different (more european ones) combinations of auger wagons and transport trailers as well.
Not sure what might be the best solution here, whether user sets the desired pipe position manually (as was possible in FS19) or CP tries to adjust itself.

**Comment by Misamangame** [comment]
@pvaiko @rohne83 @Tensuko update : 

No it will not discharge in that position where the tractor is ending up. But this is case of Bergmann & Hawe unloaders and Wilson trailer. Bergmann can unload to Lodeking to part which is empty (1st, 2nd or 3rd) but then also CP disengage when he fills up only that 1 part. I saw some video where unloader reversed, but it is not a case here.  with HAWE i had same result like with Wilson trailer only difference he managed to discharge few litres before CP disengaged.

I guess what @rohne83 is saying is correct I think generally all auger wagons should line up a bit further away from trailers as they seems way too close. 

I can also test american augers later

**Comment by Tensuko** [comment]
Further away wont work for some. This is testing experience and trust me I tested a lot...
Finding something that works the best for almost all wadnt easy.

**Comment by Misamangame** [comment]
@Tensuko I believe you. I know it will never be perfect and not looking for perfection. That should be Giants job to make a game much better then is. I know those american trailers have problems with AD unloading into them too and find some proper position for different kind of augers was hard. I always stick with what i have and if it means not to use those trailers for this job I have no problem with that. I just reported what I found :) Generally CP unloader work great even on my big fields so I'm happy with it.
